presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1159,8 +1165,15 @@ protected Scope visitSetOperation(SetOperation node, Optional<Scope> scope)
             Type[] outputFieldTypes = relationScopes.get(0).getRelationType().getVisibleFields().stream()
                     .map(Field::getType)
                     .toArray(Type[]::new);
+            int outputFieldSize = outputFieldTypes.length;
+            if (isExpensiveDistinctUnion(node, outputFieldTypes)) {
```

Comment:
What about naming this function `isExpensiveUnionDistinct` to be consistent with the SQL?